deepspeed/runtime/zero/partition_parameters.py

```diff
@@ -3,29 +3,68 @@
 Licensed under the MIT license.
 """
 
+import math
 import os
 import time
 import types
+from typing import Callable, Iterable
 from enum import Enum
 import functools
 import itertools
+from typing import List
 
 import torch
-from torch.distributed.distributed_c10d import _get_global_rank, group
+from torch import Tensor
 import torch.distributed as dist
+from torch.distributed.distributed_c10d import _get_global_rank, group
+from torch.nn import Module
+from torch.nn import Parameter
 
 from .linear import LinearModuleForZeroStage3, LinearFunctionForZeroStage3
 from .offload_constants import *
 
-from ..utils import see_memory_usage
-from deepspeed.utils import log_dist, init_distributed, logger
+from ..utils import get_only_unique_item, see_memory_usage
+from deepspeed.runtime.zero.utils import assert_ints_same_as_other_ranks
+from deepspeed.utils import init_distributed, instrument_w_nvtx, logger
 from deepspeed.utils.debug import debug_param2name_id_shape, debug_param2name_id_shape_device, debug_module2name, debug_param2name, debug_param2name_id_shape_status, printflock, log_rank_file
+from deepspeed.utils.logging import logger
 
 from ..swap_tensor.partitioned_param_swapper import AsyncPartitionedParameterSwapper, PartitionedParamStatus
 from ..config import DeepSpeedConfig
 
 param_count = 0
-partitioned_param_data_shape = [1]
+partitioned_param_data_shape = [0]
```

Comment:
Looks like `partitioned_param_data_shape` is not originally used. Is there a reason for this change?

Reply:
hm, no might just be vestigial from something i was doing earlier. will revert this

Reply:
since its not used do you mind if i remove it to avoid confusing readers? this PR changes the shape of partitioned parameters to (0,) instead of (1,)

Reply:
Sure, please remove

Reply:
oh nvm, its actually used but only in unit tests. so its actually required to have
`partitioned_param_data_shape = [0]` otherwise the tests fail